cpp/src/text/detokenize.cu

```diff
@@ -155,42 +139,27 @@ std::unique_ptr<cudf::column> detokenize(cudf::strings_column_view const& string
     return cudf::make_empty_column(cudf::data_type{cudf::type_id::STRING});
 
   auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
-  // the indices may not be in order so we need to sort them
-  auto sorted_rows     = cudf::stable_sorted_order(cudf::table_view({row_indices}));
+  // the indices may not be in order so we need to build a sorted map
+  auto sorted_rows = cudf::detail::stable_sorted_order(
+    cudf::table_view({row_indices}), {}, {}, stream, rmm::mr::get_current_device_resource());
   auto const d_row_map = sorted_rows->view().data<int32_t>();
 
   // create offsets for the tokens for each output string
   auto tokens_offsets =
-    cudf::type_dispatcher(row_indices.type(),
-                          token_row_offsets_fn{row_indices, sorted_rows->view(), tokens_counts},
-                          stream);
-  auto const output_count = tokens_offsets->size() - 1;  // number of output strings
+    create_token_row_offsets(row_indices, sorted_rows->view(), tokens_counts, stream);
+  auto const output_count = tokens_offsets.size() - 1;  // number of output strings
 
-  // create output strings offsets by calculating the size of each output string
   cudf::string_view const d_separator(separator.data(), separator.size());
-  auto offsets_transformer_itr = thrust::make_transform_iterator(
-    thrust::make_counting_iterator<cudf::size_type>(0),
-    detokenizer_fn{*strings_column, d_row_map, tokens_offsets->data(), d_separator});
-  auto offsets_column = cudf::strings::detail::make_offsets_child_column(
-    offsets_transformer_itr, offsets_transformer_itr + output_count, stream, mr);
-  auto d_offsets = offsets_column->view().data<int32_t>();
-
-  // build the chars column - append each source token to the appropriate output row
-  cudf::size_type const total_bytes =
-    cudf::detail::get_value<int32_t>(offsets_column->view(), output_count, stream);
-  auto chars_column = cudf::strings::detail::create_chars_child_column(total_bytes, stream, mr);
-  auto d_chars      = chars_column->mutable_view().data<char>();
-  thrust::for_each_n(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator<cudf::size_type>(0),
+
+  auto children = cudf::strings::detail::make_strings_children(
+    detokenizer_fn{*strings_column, d_row_map, tokens_offsets.data(), d_separator},
     output_count,
-    detokenizer_fn{
-      *strings_column, d_row_map, tokens_offsets->data(), d_separator, d_offsets, d_chars});
-  chars_column->set_null_count(0);
+    stream,
+    mr);
 
   // make the output strings column from the offsets and chars column
   return cudf::make_strings_column(
-    output_count, std::move(offsets_column), std::move(chars_column), 0, rmm::device_buffer{});
+    output_count, std::move(children.first), std::move(children.second), 0, rmm::device_buffer{});
```

Comment:
I remember that we prefer `rmm::device_buffer{0, stream, mr}`.

Reply:
At one time we did. But we have now shifted back to this `rmm::device_buffer{}`.